packages/utils/src/date-utils/timeFormatting.ts

```diff
@@ -71,11 +70,80 @@ export function removeSeconds(timer: string): string {
 }
 
 /**
- * @description utility function to format a date in milliseconds using luxon
- * @param {number} millis
- * @param {string} format
- * @return {string}
+ * Formats a given date into a custom string format based on UTC time.
+ * 
+ * @param millis - The number of milliseconds.
+ * @param format - A string specifying the desired output format. 
+ *                 For example, 'yyyy-MM-dd HH:mm:ss' will format the date as '2024-01-23 09:41:08'.
+ * 
+ * @returns The formatted date as a string according to the provided `format` string.
+ * 
+ * @throws Will throw an error if the date formatting fails or the regex match returns null, which should not happen
+ *         if the `Intl.DateTimeFormat` is correctly configured and the input `millis` is valid.
+ * 
+ * @throws Will throw an error if the input @param millis is smaller than zero.
  */
 export function formatFromMillis(millis: number, format: string): string {
-  return DateTime.fromMillis(millis).toUTC().toFormat(format);
+  if (millis < 0) {
+    throw new Error("Input `millis` can't be smaller than zero.");
```

Comment:
I would say, that the best way forward is to return a falsy value (undefined), because wrapping every consumer in a try-catch would lead to a chore, even though it would be better for clean code.